interfaces/builtin/basedeclaration.go

```diff
@@ -138,6 +138,9 @@ authority-id: canonical
 series: 16
 revision: 0
 plugs:
+  classic-dimension:
+    allow-installation: false
+    deny-auto-connection: true
   docker-support:
     allow-installation: false
     deny-auto-connection: true
```

Comment:
Whoops! We do still need the slot side to say this is ok when provided by the core snap. Sorry about not remembering this in the initial review. Please, in addition to the above, add:
```
slots:
  ...
  classic-dimension:
    allow-installation:
      slot-snap-type:
        - core
    deny-auto-connection: true
```